src/components/Comments.js

```diff
@@ -0,0 +1,253 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Button, Col, Row, Glyphicon } from 'react-bootstrap';
+
+import QueryResponse from '../api/QueryResponse';
+import AuthUtils from '../util/AuthUtils';
+import SimpleQueryRequest from '../api/SimpleQueryRequest';
+
+type CommentsProps = {
+  docId: string,
+  commentsTable: string | null,
+}
+
+type CommentsDefaultProps = {
+  commentsTable: string,
+}
+
+type CommentsState = {
+  comment: string;
+  response: QueryResponse | null;
+  error: string | null;
+  showCommentModal: boolean,
+  commentList: Array<any>,
+}
+
+class Comments extends React.Component<CommentsDefaultProps, CommentsProps, CommentsState> {
+  static contextTypes = {
+    searcher: PropTypes.any,
+  };
+
+  static defaultProps = {
+    commentsTable: 'comments',
+  };
+
+  static displayName = 'Comments';
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      comment: '',
+      response: null,
+      error: null,
+      showCommentModal: false,
+      commentList: [],
+    };
+    (this: any).showCommentModal = this.showCommentModal.bind(this);
+    (this: any).hideCommentModal = this.hideCommentModal.bind(this);
+    (this: any).captureComment = this.captureComment.bind(this);
+    (this: any).saveComment = this.saveComment.bind(this);
+    (this: any).clearComment = this.clearComment.bind(this);
+    (this: any).getComments = this.getComments.bind(this);
+    (this: any).deleteComment = this.deleteComment.bind(this);
+  }
+
+  componentWillMount() {
```

Comment:
Use `componentDidMount` instead of `componentWillMount`.
https://reactjs.org/docs/react-component.html#mounting